File: javascript/space-age/space-age.js
const periodMap = new Map([
  ["mercury", 0.2408467],
  ["venus", 0.61519726],
  ["earth", 1.0],
  ["mars", 1.8808158],
  ["jupiter", 11.862615],
  ["saturn", 29.447498],
  ["uranus", 84.016846],
  ["neptune", 164.79132],
]);

const EARTH_YEAR_IN_SECONDS = 31_557_600;

export const age = (planet, seconds) => +(seconds / EARTH_YEAR_IN_SECONDS / periodMap.get(planet)).toFixed(2);

File: scala/space-age/src/main/scala/SpaceAge.scala
import scala.language.dynamics

object SpaceAge extends Dynamic {

  val periodMap = Map(
    "Mercury" -> 0.2408467,
    "Venus" -> 0.61519726,
    "Earth" -> 1.0,
    "Mars" -> 1.8808158,
    "Jupiter" -> 11.862615,
    "Saturn" -> 29.447498,
    "Uranus" -> 84.016846,
    "Neptune" -> 164.79132
  )

  val EARTH_YEAR_IN_SECONDS = 31557600

  def onPlanet(planet: String, seconds: Double): Double = seconds / EARTH_YEAR_IN_SECONDS / periodMap(planet)

  // onMercury, onVenus, ... -> extract planet name using substring
  def applyDynamic(methodName: String)(seconds: Double): Double = onPlanet(methodName.substring(2), seconds)

}
